components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go

```diff
@@ -79,6 +94,15 @@ type Condition struct {
 // FunctionStatus defines the observed state of FuncSONPath: .status.phase
 type FunctionStatus struct {
 	Conditions []Condition `json:"conditions,omitempty"`
+	Repository `json:",inline,omitempty"`
```

Comment:
do we want this info in every function? shouldn't it be a pointer if it's not needed everywhere?

Reply:
Could you please help me understand the what do you mean? 
- in a non git type function we will get e.g. 
```
...
Status:
  Conditions:
    Last Transition Time:  2020-07-28T11:06:09Z
    Message:               Deployment demo-57vkq is not ready yet
    Reason:                DeploymentWaiting
    Status:                Unknown
    Type:                  Running
    Last Transition Time:  2020-07-28T11:06:07Z
    Message:               Job demo-build-8t8zq finished
    Reason:                JobFinished
    Status:                True
    Type:                  BuildReady
    Last Transition Time:  2020-07-28T11:05:54Z
    Message:               ConfigMap demo-thlxf created
    Reason:                ConfigMapCreated
    Status:                True
    Type:                  ConfigurationReady
  Source:                  module.exports = {
    main: function(event, context) {
      return 'Hello World'
    }
  }
```
- in git type function we will get e.g.
```
...
Status:
  Base Dir:  /
  Branch:    master
  Commit:    2e97a97079e2a41ab794d77fe1ccdc46d76e71db
  Conditions:
    Last Transition Time:  2020-07-28T11:13:49Z
    Message:               Job newone-build-tzjwz is still in progress
    Reason:                JobRunning
    Status:                Unknown
    Type:                  BuildReady
    Last Transition Time:  2020-07-28T11:13:46Z
    Message:               Sources newone updated
    Reason:                SourceUpdated
    Status:                True
    Type:                  ConfigurationReady
  Dockerfile:              deploy/Dockerfile
  Source:                  https://github.com/pPrecel/public-gitops
```

Reply:
Ok, I misunderstood how it works, I thought that you just need a pointer + "omitempty" combo to achive that